### PR TITLE
feat: add search in projects

### PR DESCRIPTION
**Pull request by ykhedher**

### CODE REVIEW

**Comment by ykhedher** [comment]
[project search](https://trello.com/c/bbFsqn5w/913-project-search)

File: src/component/project/ProjectList/ProjectList.styles.ts [hunk 1]
```diff
@@ -21,4 +21,20 @@ export const useStyles = makeStyles(theme => ({
...
+        display: 'flex',
+        gap: '1rem',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        [theme.breakpoints.down('xs')]: {
+            display: 'block',
+        },
+    },
+    searchBar: {
+        minWidth: '450px',
```
Comment (olav) [inline src/component/project/ProjectList/ProjectList.styles.ts:35]: `450`

File: src/component/project/ProjectList/ProjectList.tsx [hunk 2]
```diff
@@ -43,14 +43,22 @@ function resolveCreateButtonData(isOss: boolean, hasAccess: boolean) {
...
     const { projects, loading, error, refetch } = useProjects();
     const [fetchedProjects, setFetchedProjects] = useState<projectMap>({});
     const ref = useLoading(loading);
     const { isOss } = useUiConfig();
+    const [filter, setFilter] = useState('');
+
+    const filteredProjects = useMemo(() => {
+        const regExp = new RegExp(filter, 'i');
+        return filter
+            ? projects?.filter(project => regExp.test(project?.name))
```
Comment (olav) [inline src/component/project/ProjectList/ProjectList.tsx:59]: Looks like `projects` will always be defined, so we don't need the `?`. Same for `project?.name`.

File: src/component/project/ProjectList/ProjectList.tsx [hunk 3]
```diff
@@ -82,7 +90,7 @@ const ProjectListNew = () => {
             return renderLoading();
         }
 
-        return projects.map((project: IProjectCard) => {
+        return filteredProjects?.map((project: IProjectCard) => {
```
Comment (olav) [inline src/component/project/ProjectList/ProjectList.tsx:93]: -`?`

File: src/component/project/ProjectList/ProjectList.tsx [hunk 4]
```diff
@@ -148,7 +166,7 @@ const ProjectListNew = () => {
                 <ConditionallyRender condition={error} show={renderError()} />
                 <div className={styles.container}>
                     <ConditionallyRender
-                        condition={projects.length < 1 && !loading}
+                        condition={filteredProjects?.length < 1 && !loading}
```
Comment (olav) [inline src/component/project/ProjectList/ProjectList.tsx:169]: -`?`

**Review by olav — CHANGES_REQUESTED** [review]

**Review by olav — APPROVED** [review]

**Pull request merged**